src/neuroglancer/datasource/computed/backend.ts

```diff
@@ -0,0 +1,508 @@
+    const originSource = this.outputSource_!.originSource;
+    const originChunkSize = originSource.spec.chunkDataSize;
+    const inputSpec = this.computationParams_!.inputSpec;
+    const inputLower = this.inputLower_!;
+    const gridLower =
+        vec3.floor(vec3.create(), vec3.divide(vec3.create(), inputLower, originChunkSize));
+    const inputSizeMinusOne = vec3.subtract(vec3.create(), inputSpec.size, [1, 1, 1]);
+    const inBoxUpper = vec3.add(vec3.create(), inputLower, inputSizeMinusOne);
+    vec3.max(gridLower, gridLower, [0, 0, 0]);
+    vec3.min(inBoxUpper, inBoxUpper, originSource.spec.upperVoxelBound);
+    const gridUpper =
+        vec3.floor(vec3.create(), vec3.divide(vec3.create(), inBoxUpper, originChunkSize));
+
+    const gridPosition = vec3.create();
+    for (let z = gridLower[2]; z <= gridUpper[2]; ++z) {
+      for (let y = gridLower[1]; y <= gridUpper[1]; ++y) {
+        for (let x = gridLower[0]; x <= gridUpper[0]; ++x) {
+          gridPosition.set([x, y, z]);
+          const key = gridPositionKey(gridPosition);
+          this.originGridPositions_.set(key, vec3.copy(vec3.create(), gridPosition));
+        }
+      }
+    }
+
+    for (const chunkGridPosition of this.originGridPositions_.values()) {
+      const chunk = originSource.getChunk(chunkGridPosition);
+      chunk.registerListener(this);
+      this.stateChanged(chunk);
+    }
+  }
+}
+
+@registerSharedObject() export class ComputedVolumeChunkSource extends
+(WithParameters(VolumeChunkSource, ComputedVolumeChunkSourceParameters)) {
+  originSource: VolumeChunkSource;
+  private activeComputations_ = new Map<String, ComputedVolumeChunk>();
+  private frontendChunkRequests_ =
+      new Map<String, Map<String, (array: ArrayType, error: any) => void>>();
+  public computation: VolumeComputationBackend;
+
+  constructor(rpc: RPC, options: any) {
+    super(rpc, options);
+    this.originSource = this.rpc!.getRef<VolumeChunkSource>(this.parameters.sourceRef);
+    this.computation = this.rpc!.getRef<VolumeComputationBackend>(this.parameters.computationRef);
+    this.registerDisposer(this.chunkManager);
+    this.registerDisposer(this.chunkManager.recomputeChunkPrioritiesLate.add(() => {
+      this.updateChunkPriorities();
+    }));
+  }
+
+  updateChunkPriorities() {
+    const priorityMap = new Map<String, {priority: number, tier: number, pos: vec3}>();
+
+    for (const outputChunk of this.activeComputations_.values()) {
+      for (const gridPosition of outputChunk.getOverlappingOriginGridPositions()) {
+        const gridString = gridPosition.toLocaleString();
+        if (!priorityMap.has(gridString)) {
+          priorityMap.set(
+              gridString,
+              {priority: outputChunk.priority, tier: outputChunk.priorityTier, pos: gridPosition});
+        } else {
+          const currentPriority = priorityMap.get(gridString)!;
+          currentPriority.priority = Math.max(currentPriority.priority, outputChunk.priority);
```

Comment:
This logic isn't quite right, because you can't max/min tier and priority independently.  However, requestChunk already does this min/max logic so you can just use that directly rather than building priorityMap

Reply:
Fixed.